mlflow/artifacts/__init__.py

```diff
@@ -57,3 +59,107 @@ def download_artifacts(
+
+        import mlflow
+        with mlflow.start_run() as run:
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_text("This is a sentence", "file.txt")
+            file_content = mlflow.artifacts.load_text(artifact_uri + "/file.txt")
+            print(file_content)
+
+    .. code-block:: text
+        :caption: Output
+
+        This is a sentence
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        with open(local_artifact) as local_artifact_fd:
+            try:
+                return str(local_artifact_fd.read())
+            except Exception:
+                raise MlflowException("Unable to form a str object from file content", BAD_REQUEST)
+
+
+def load_dict(artifact_uri: str) -> dict:
+    """
+    Loads content of the artifact file as json/dict object
+
+    :param artifact_uri: artifact location
+    :return: dict
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        with mlflow.start_run() as run:
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_dict({"mlflow-version": "0.28", "n_cores": "10"}, "config.json")
+            config_json = mlflow.artifacts.load_dict(artifact_uri + "/config.json")
+            print(config_json)
+
+    .. code-block:: text
+        :caption: Output
+
+        {'mlflow-version': '0.28', 'n_cores': '10'}
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        with open(local_artifact) as local_artifact_fd:
+            try:
+                return json.load(local_artifact_fd)
+            except json.JSONDecodeError:
+                raise MlflowException("Unable to form a JSON object from file content", BAD_REQUEST)
+
+
+def load_image(artifact_uri: str):
+    """
+    Loads content of the artifact file as PIL.Image object
+
+    :param artifact_uri: artifact location
+    :return: PIL.Image
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        from PIL import Image
+        with mlflow.start_run() as run:
+            image = Image.new("RGB", (100, 100))
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_image(image, "image.png")
+            image  = mlflow.artifacts.load_image(artifact_uri + "/image.png")
+            print(image)
+
+    .. code-block:: text
+        :caption: Output
+
+        <PIL.PngImagePlugin.PngImageFile image mode=RGB size=100x100 at 0x11D2FA3D0>
+    """
+    try:
+        from PIL import Image
+    except ImportError as exc:
+        raise ImportError(
+            "`load_image` requires Pillow. Please install it via: pip install Pillow"
+        ) from exc
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        try:
+            image_obj = Image.open(local_artifact)
```

Comment:
```suggestion
            image_obj = Image.open(local_artifact)
            image_obj.load()
```

https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.open:

> This is a lazy operation; this function identifies the file, but the file remains open and the actual image data is not read from the file until you try to process the data (or call the [load()](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.load) method).